src/EditorFeatures/CSharpTest/MisplacedUsings/MisplacedUsingsCodeFixProviderTests.cs

```diff
@@ -0,0 +1,1295 @@
+        [InlineData(StructDefinition)]
+        [InlineData(InterfaceDefinition)]
+        [InlineData(EnumDefinition)]
+        [InlineData(DelegateDefinition)]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithTypeDefinition_ValidUsingStatements(string typeDefinition)
+        {
+            var testCode = $@"using System;
+
+{typeDefinition}
+";
+
+            return VerifyAnalyzerAsync(testCode, s_insidePreferenceOption, DiagnosticResult.EmptyDiagnosticResults);
+        }
+
+        /// <summary>
+        /// Verifies that having using statements in the compilation unit will not produce any diagnostics when there are attributes present.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithAttributes_ValidUsingStatements()
+        {
+            var testCode = @"using System.Reflection;
+
+[assembly: AssemblyVersion(""1.0.0.0"")]
+
+namespace TestNamespace
+{
+    using System;
+    using System.Threading;
+}
+";
+
+            return VerifyAnalyzerAsync(testCode, s_insidePreferenceOption, DiagnosticResult.EmptyDiagnosticResults);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will properly reorder using statements when placing System first.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnit_UsingsMovedWithSystemPlacedFirst()
+        {
+            var testCode = @"using Microsoft.CodeAnalysis;
+using SystemAction = System.Action;
+using static System.Math;
+using System;
+
+using static System.String;
+using MyFunc = System.Func<int,bool>;
+
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Foo
+{
+    public class Bar
+    {
+    }
+}
+";
+
+            var fixedTestCode = @"namespace Foo
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using Microsoft.CodeAnalysis;
+    using static System.Math;
+    using static System.String;
+    using MyFunc = System.Func<int,bool>;
+    using SystemAction = System.Action;
+
+    public class Bar
+    {
+    }
+}
+";
+
+            DiagnosticResult[] expected =
+            {
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(1, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(2, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(3, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(4, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(6, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(7, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(9, 1),
+                Diagnostic(MisplacedUsingsDiagnosticAnalyzer._insideDescriptor).WithLocation(10, 1),
```

Comment:
1. why are we reporting so many misplaced diagnostics?  Users should really only see one diagnostic (either for the span of the first item, or perhaps the span of the entire set).
2. I don't like this test because it's no longer declarative.  i have no idea what `10,1` means, or how it relates to the code.